build-coredistools.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+RootDirectory="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+SourcesDirectory=$RootDirectory/src
+BinariesDirectory=$RootDirectory/obj
+TargetOSArchitecture=$1
+CrossRootfsDirectory=$2
+
+case "$TargetOSArchitecture" in
+    linux-arm)
+        CrossCompiling=1
+        TargetTriple=arm-linux-gnueabihf
+        ;;
+
+    linux-arm64)
+        CrossCompiling=1
+        TargetTriple=aarch64-linux-gnu
+        ;;
+
+    linux-x64|macos-x64)
+        CrossCompiling=0
+        ;;
+
+    *)
+        echo "Unknown target OS and architecture: $TargetOSArchitecture"
```

Comment:
A similar comment applies here about it being useful to have a "help" / usage comment and, if necessary and separate, a "what does this script do" header comment.